projects/graph/social/social.py

```diff
@@ -73,16 +75,26 @@ def getAllSocialPaths(self, userID):
 
         The key is the friend's ID and the value is the path.
         """
-        visited = {}  # Note that this is a dictionary, not a set
         # !!!! IMPLEMENT ME
-        # queue = []
-        # queue.append(userID)
-        # while len(queue) > 0:
-        #     node = queue.pop()
-        #     visited[node] = node
-        #     for friends in self.users[node]:
-        #         if friends not in visited:
-        #             queue.append(friends)
+        visited = {userID: [userID]}
+        queue = Queue()
+        queue.enqueue([userID])
+
+        while queue.len() > 0:
+
+            path = queue.dequeue()
+
+            node = path[-1]
+
+            for friend in self.friendships[node]:
+                if friend not in visited:
+
+                    upath = path + [friend]
+
+                    visited[friend] = upath
+
+                    queue.enqueue(upath)
+
```

Comment:
This is excellent work Michael. I'm very happy to see that you went with this implementation strategy for you BFS. I think by adding in these structures you can really start to understand how generalizable these methods really are. All you needed to do here was repeatedly visit the nodes around the passed in node, and then follow the rest of their branches and save the paths. 
This is top notch work and a fantastic turnaround around form last week! I'm really excited to see you make so much improvement.